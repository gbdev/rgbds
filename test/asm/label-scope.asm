ASSERT !DEF(@) && !DEF(.)

PURGE @, .

SECTION "test", ROM0[42]
Foobar:

PURGE @, .

ASSERT DEF(@) && DEF(.) && DEF(Foobar)

PRINTLN "PC: {#05X:@}; label scope: \"{.}\"; {.}: {#05X:{.}}"
