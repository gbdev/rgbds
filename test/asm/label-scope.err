error: label-scope.asm(3):
    '@' not defined
error: label-scope.asm(3):
    '.' not defined
error: label-scope.asm(8):
    Built-in symbol '@' cannot be purged
error: label-scope.asm(8):
    Built-in symbol '.' cannot be purged
error: Assembly aborted (4 errors)!
