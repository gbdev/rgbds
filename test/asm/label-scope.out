PC: $002A; label scope: "Foobar"; Foobar: $002A
